app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,26 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user_id = current_user.id
+    if @post.save
+      flash[:success] = 'Post created!'
+      redirect_to user_post_path(current_user, @post)
+    else
+      flash.now[:error] = 'Post not created!'
+      render :new
+    end
+  end
```

Comment:
- [ ] I got an error when I tried to create a post on behalf of a current user. Kindly fix this error in order to move forward.

![user](https://user-images.githubusercontent.com/53124548/214413878-f74d1b64-0f34-447e-bdb1-d3a295b193eb.png)